apps/prairielearn/src/pages/home/home.sql

```diff
@@ -200,6 +206,55 @@ ORDER BY
   d.end_date DESC NULLS LAST,
   ci.id DESC;
 
+-- BLOCK select_student_courses_modern_publishing
+SELECT
+  c.short_name AS course_short_name,
+  c.title AS course_title,
+  to_jsonb(ci) AS course_instance,
+  to_jsonb(e) AS enrollment,
+  to_jsonb(cie) AS latest_publishing_extension
+FROM
+  enrollments AS e
+  LEFT JOIN users AS u ON (u.user_id = e.user_id)
+  JOIN course_instances AS ci ON (
+    ci.id = e.course_instance_id
+    AND ci.deleted_at IS NULL
+    -- We only consider courses using the modern publishing system in this query.
```

Comment:
This query isn't evaluating access in the function, that filtering happens in typescript. It should be a small set since it only considers instances linked to your enrollment.